src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -633,8 +633,11 @@ private Conversion DeriveStandardExplicitFromOppositeStandardImplicitConversion(
                 case ConversionKind.ImplicitReference:
                     impliedExplicitConversion = Conversion.ExplicitReference;
                     break;
-                case ConversionKind.Boxing:
-                    impliedExplicitConversion = Conversion.Unboxing;
+                case ConversionKind.ValueTypeBoxing:
+                    impliedExplicitConversion = Conversion.ValueTypeBoxing;
+                    break;
+                case ConversionKind.TypeParameterBoxing:
+                    impliedExplicitConversion = Conversion.TypeParameterBoxing;
```

Comment:
Yes.  This code looks just terribly broken.  Is there a way i could write a test to validate that this is working properly?